stan/math/rev/core/deep_copy_vars.hpp

```diff
@@ -0,0 +1,87 @@
+#ifndef STAN_MATH_REV_CORE_DEEP_COPY_VARS_HPP
+#define STAN_MATH_REV_CORE_DEEP_COPY_VARS_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/rev/meta.hpp>
+#include <stan/math/rev/core/var.hpp>
+
+#include <utility>
+#include <vector>
+
+namespace stan {
+namespace math {
+
+/**
+ * Forward arguments that do not contain vars. There
+ *   is no copying to be done.
+ *
+ * @tparam Arith an arithmetic type.
+ * @param arg For lvalue references this will be passed by reference.
+ *  Otherwise it will be moved.
+ */
+template <typename Arith, typename = require_arithmetic_t<scalar_type_t<Arith>>>
+inline decltype(auto) deep_copy_vars(Arith&& arg) {
```

Comment:
What do we get if the tuple passed into the function is an empty one? Does this function then qualify? If yes, then can we please test it - I think we don't do that atm.

Reply:
I added an empty tuple test to the apply function.

I think this is maybe a higher level question that deep copy. I think it's covered by this reduce_sum test (what happens when there are no local args): https://github.com/stan-dev/math/blob/6743ff6937159857b1409f8ee15c7c9dc7589670/test/unit/math/mix/functor/reduce_sum_test.cpp#L63